packages/zone/src/make-once.js

```diff
@@ -0,0 +1,74 @@
+// @ts-check
+const { Fail } = assert;
+
+/**
+ * @param {string} baseLabel
+ * @param {import('.').Stores} stores
+ * @param {MapStore<string, any>} [backingStore]
+ */
+export const makeOnceKit = (baseLabel, stores, backingStore) => {
+  const usedKeys = stores.detached().setStore(`${baseLabel} used keys`);
```

Comment:
The `detached()` here is awesome!

Reply:
It took me a minute to realize why detached was important. Maybe add a comment with details?